core-bundle/src/Twig/Finder/Finder.php

```diff
@@ -123,14 +122,34 @@ public function asTemplateOptions(): array
             return $this->translator->trans('MSC.global', [], 'contao_default');
         };
 
-        foreach (array_keys(iterator_to_array($this->getIterator())) as $identifier) {
-            $label = sprintf(
-                '%s [%s]',
+        $getCustomLabel = function (string $identifier, array $sourceLabels): string|null {
+            $translationId = "TEMPLATE.$identifier";
+
+            if (!$this->translator->getCatalogue()->has($translationId, 'contao_default')) {
```

Comment:
I’m not sure if we should introduce a new `TEMPLATE` category here that is maped to the `default` language file. We would have to add it here for example: https://github.com/contao/contao/blob/3df28e79dfaddcdf403023e12b033d7a27a68cd4/core-bundle/contao/library/Contao/InsertTags.php#L360-L363

But if we would use lowercase `template` the `template.php` translation file would be loaded automatically.

```suggestion
            $translationId = "template.$identifier";

            if (!$this->translator->getCatalogue()->has($translationId, 'contao_template')) {
```

Reply:
I went with a new translation domain instead of a prefix in https://github.com/contao/contao/pull/5717#issuecomment-1403472654.